### ISSUE TITLE
[Streaming]: Doorbell notification live stream not starting on Apple Watch If no cached snapshot is found

### ISSUE THREAD

**Issue by malmentm** [dgreif/ring#861]
### Streaming Issue

Doorbell notification live stream not starting on Apple Watch If no cached snapshot is found. I have made some testing and as long as there is att cached snapshot for the doorbell the steam starts fine on the watch but when no cached snapshot is found the video stream is not started on the watch and you get is a spinning wheel. Starting stream manually from watch always works I I use the home app on the watch.

### Proposed Solution

Always fetch last known cached snapshot maybe a solution? Or do these cached snapshots expire?

### Environment

Homebridge

### What operating system are you on?

Mac

### Relevant log output

```shell
This is a log entry from when the stream does not start
___
[25/01/2022, 23:45:02] [Ring] Front Door Doorbell Pressed
[25/01/2022, 23:45:02] [Ring] No snapshot cached for Front Door
[25/01/2022, 23:45:02] [Ring] Loading new snapshot into cache for Front Door
[25/01/2022, 23:45:02] [Ring] No snapshot cached for Front Door
[25/01/2022, 23:45:02] [Ring] No snapshot cached for Front Door
[25/01/2022, 23:45:39] [Ring] Snapshot cached for Front Door (37.198s)
___

This is a log entry from where the stream starts just fine

___
[25/01/2022, 23:46:29] [Ring] Front Door Detected Motion
[25/01/2022, 23:46:32] [Ring] Front Door Doorbell Pressed
[25/01/2022, 23:46:32] [Ring] Used cached snapshot for Front Door
[25/01/2022, 23:46:32] [Ring] Loading new snapshot into cache for Front Door
[25/01/2022, 23:46:32] [Ring] Used cached snapshot for Front Door
[25/01/2022, 23:46:32] [Ring] Used cached snapshot for Front Door
[25/01/2022, 23:46:36] [Ring] Preparing Live Stream for Front Door
[25/01/2022, 23:46:36] [Ring] Connecting to Front Door using STUN
[25/01/2022, 23:46:36] [Ring] Waiting for stream data from Front Door (0.327s)
----
```

### Honesty Time

- [X] I swear this is worth your time!

**Comment by malmentm** [comment]
It seems that those times there's no cached snapshot the is also no preview picture on the phones notification. So the conclusion seems to be if there is no cached snapshot the video notification on the watch will not start either. 

Is there som way to always have a cached snapshot available?

**Comment by malmentm** [comment]
Another observation is that if I open the home app on the Mac the snapshot is black until a new snapshot is fetched.

[26/01/2022, 09:42:29] [Ring] No snapshot cached for Front Door
[26/01/2022, 09:42:29] [Ring] Loading new snapshot into cache for Front Door
[26/01/2022, 09:42:29] [Ring] No snapshot cached for Front Door
[26/01/2022, 09:42:34] [Ring] Snapshot cached for Front Door (4.437s)
[26/01/2022, 09:42:39] [Ring] Used cached snapshot for Front Door

**Comment by malmentm** [comment]
This is how it looks when I get a black snapshot on Mac

[26/01/2022, 20:26:47] [Ring] No snapshot cached for Front Door
[26/01/2022, 20:26:47] [Ring] Loading new snapshot into cache for Front Door
[26/01/2022, 20:26:56] [Ring] Snapshot cached for Front Door (8.925s)
[26/01/2022, 20:27:07] [Ring] Used cached snapshot for Front Door

**Comment by malmentm** [comment]
I think this might be related to this post. https://github.com/dgreif/ring/issues/825

**Comment by malmentm** [comment]
@dgreif Would it be possible to always keep the last snapshot in cache? Even if it's very old? I think that would make the experience better.

**Comment by malmentm** [comment]
I'have tried to enable the "Avoid snapshot battery drain" and this solves the problem until the 10 minutes window expires. A solution to the problem might be to always provide the last cached snapshop (even if very old) on first request (instead of the empty image you said was used in (https://github.com/dgreif/ring/issues/825) . And after that request a new snapshop from ring and load into cache. After 10 seconds the home app will request a new snapshop and the latest snapshot will be loaded.

**Comment by malmentm** [comment]
Another maybe better option might be to fetch snapshot when option is detected?

**Comment by malmentm** [comment]
After some testing the cause of this must be the black preview image that often appears on MacOS and somtimes on the iOS device. If the below behavior was implemented i'am certain the problem will dissapear.

A solution to the problem might be to always provide the last cached snapshop (even if very old) on first request (instead of the empty image you said was used in (https://github.com/dgreif/ring/issues/825) . And after that request a new snapshop from ring and load into cache. After 10 seconds the home app will request a new snapshop and the latest snapshot will be loaded.

**Comment by malmentm** [comment]
@dgreif Is it possible to add this change of cached snapshot behavior? I tried v10.0.0-beta.0 and is has the same issue with black snapshot when no fresh cached snapshot is found.

**Comment by dgreif** [comment]
Should be fixed in 10.0.0-beta.4
